### ISSUE TITLE
experimental_ncon is done

### ISSUE THREAD

**Issue by alewis** [google/TensorNetwork#774]
Once PRs #775 and #767 are merged, and the latter merged into experimental_ncon, experimental_ncon will to my mind be complete. We should thus begin thinking about how to merge it into master. The branch is somewhat misleadingly named since the main new thing about it is the tn.Tensor class. 

I've written a vuMPS implementation in terms of TensorNetwork via tn.Tensor in https://github.com/alewis/tn_vumps. There's still some cleaning up to do, but it's basically done. This demonstrates what to my mind are the significant advantages of this class:

Compared to directly using a backend object:
- To me, the syntax looks simpler.
- The backend is an extra concept that the user no longer needs to understand.
- Since the backend is a member of tn.Tensor, it can be easily localized within functions without any extra arguments to be passed.

Compared to directly using a backend object, or to writing backend independent functions that directly operate on the backend's native array class:
- Better error handling in cases of backend clash.
- Attributes of tn.Tensor behave identically as the same function in operator form (e.g. A.ravel and tn.ravel(A) are the same code).
- Matvec functions in Krylov methods can be written easily in a backend independent way without resorting to the backend module directly. E.g. passing lambda x: A @ x where x and A are tn.Tensor into eigsh will just work, even with Jit, without extra traces.

Compared to not using TensorNetwork at all:
- It is very easy to switch between NumPy CPU code and Jax GPU/TPU code. 
- We provide a number of extra conveniences that simplify life and spark joy. For example the pivot_axis arguments to QR make canonicalizing an MPS tensor very simple.
- BlockSparse, but this has nothing to do with tn.Tensor of course.

I forgot to add: The vuMPS demo also demonstrates what one would do in case they need to integrate TensorNetwork with some other codebase, or if they just need some functionality that they can't implement directly through our functions. This is demonstrated in polar.py, which in the "QDWH" case needs to do different things on NumPy and Jax. It's quite simple to pull the array out of tn.Tensor, operate on it, and then wrap it back in tn.Tensor for TensorNetwork's consumption.  

Personally I find the clear distinction between the backend object and ours to be an additional advantage. If you try to use, say, jax.ravel(A) where A is a tn.Tensor, you get an error, just as you do from tn.ravel(A) where A is a jax DeviceArray. To my mind this alleviates confusion without sacrificing flexibility.

Lemme know what you think

**Comment by alewis** [comment]
I've submitted the first PR adding things to master, #779
